packages/common-sdk/src/web3/transactions/transactions-processor.ts

```diff
@@ -59,47 +58,57 @@ export class TransactionProcessor {
   public constructSendTransactions(
     transactions: Transaction[],
     lastValidBlockHeight: number,
+    blockhash: string,
     parallel: boolean = true
   ): () => Promise<PromiseSettledResult<string>[]> {
-    return async () => {
-      let done = false;
-      const isDone = () => done;
+    const executeTx = async (tx: Transaction) => {
+      const rawTxs = tx.serialize();
+      return this.connection.sendRawTransaction(rawTxs, {
+        preflightCommitment: this.commitment,
+      });
+    };
 
-      // We separate the block expiry promise so that it can be shared for all the transactions
-      const expiry = checkBlockHeightExpiry(
-        this.connection,
-        lastValidBlockHeight,
-        this.commitment,
-        isDone
-      );
-      const txs = transactions.map((tx) => tx.serialize());
-      const txPromises = txs.map(async (tx) =>
-        confirmOrExpire(this.connection, tx, this.commitment, expiry)
-      );
-      let results: PromiseSettledResult<string>[] = [];
+    const confirmTx = async (txId: string) => {
+      const result = await this.connection.confirmTransaction({
+        signature: txId,
+        lastValidBlockHeight: lastValidBlockHeight,
+        blockhash,
+      });
+
+      if (result.value.err) {
+        throw new Error(`Transaction failed: ${JSON.stringify(result.value)}`);
+      }
+    };
+
+    return async () => {
       if (parallel) {
-        results = await Promise.allSettled(txPromises);
+        const results = transactions.map(async (tx) => {
+          const txId = await executeTx(tx);
+          await confirmTx(txId);
+          return txId;
+        });
+
+        return Promise.allSettled(results);
       } else {
-        for (const txPromise of txPromises) {
-          // We might be able to have these transactions individually signed and updated, but not sure
-          // of the implications of the resigning - could be quite annoying from a user perspective
-          // if their wallet forces them to sign for each
-          results.push(await promiseToSettled(txPromise));
+        const results = [];
```

Comment:
the old implementation didn't actually send the transactions serially when `parallel=false`. it only confirmed the transactions serially